ouroboros-consensus/src/Ouroboros/Consensus/Storage/LedgerDB/DiskPolicy.hs

```diff
@@ -96,15 +88,36 @@ defaultDiskPolicy (SecurityParam k) requestedInterval = DiskPolicy {..}
          Maybe DiffTime
       -> Word64
       -> Bool
-    onDiskShouldTakeSnapshot Nothing blocksSinceLast = blocksSinceLast >= k
     onDiskShouldTakeSnapshot (Just timeSinceLast) blocksSinceLast =
-      let snapshotIntervalSeconds      = fromIntegral $ interval requestedInterval
-          snapshotInterval             = secondsToDiffTime snapshotIntervalSeconds
-          itsBeen50kBlocks             = blocksSinceLast >= 50_000
-          itsBeen6MinSinceLastSnapshot = timeSinceLast > 6 * secondsToDiffTime 60
-      in
-           timeSinceLast >= snapshotInterval
-        || itsBeen50kBlocks && itsBeen6MinSinceLastSnapshot
+         timeSinceLast >= snapshotInterval
+      || substantialAmountOfBlocksWereProcessed blocksSinceLast timeSinceLast
+
+   -- | If users never leave their wallet running for long. This would mean
+   -- that under some circumstances we would never take a snapshot
+   -- So, on startup (when the 'time since the last snapshot' is `Nothing`),
+   -- we take a snapshot as soon as there are @k@ blocks replayed.
+   -- This means that even if users frequently shut down their wallet, we still
+   -- take a snapshot roughly every @k@ blocks. It does mean the possibility of
+   -- an extra unnecessary snapshot during syncing (if the node is restarted), but
+   -- that is not a big deal.
+    onDiskShouldTakeSnapshot Nothing blocksSinceLast = blocksSinceLast >= k
```

Comment:
I would suggest moving the big comment into the body of the match. This is because, in my experience, it's easy to get confused when there are large comments between the various matches of a single function definition.

In fact, I typically favor using using one `case` instead of multiple matches for similar reasons. I think that could be a good fit here, now that you've nicely pulled out the local definitions.